### PR TITLE
system_audit: fix handling of processors yaml when user provides additional processors

### LINKED ISSUE #7042: [System Audit] Error adding processors
- **Integration:** [System Audit](https://docs.elastic.co/integrations/system_audit)
- **Integration version:** v1.5.0
- **Elastic Agent version:** 8.8.2

Adding a processor to the System Audit integration fails with this error
```
end of the stream or a document separator is expected at line 11, column 1: - add_tags: ^
```

Screenshot 2023-07-20 at 17 30 39

Seems somehow similar to https://github.com/elastic/integrations/issues/4525

### PR DESCRIPTION
**Pull request by efd6**
## What does this PR do?

When a user provides processors from the UI, the indentation will not agree with the indentation of the '- add_host_metadata' processor due to the UI widget striping leading space, and so resulting in an error.

The conventional way around this is to butt the constant sequence up against the gutter.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

- Fixes #7042

### CODE REVIEW

File: packages/system_audit/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.5.1"
+  changes:
+    - description: Fix processor UI handling.
+      type: bugfix
+      link: https://github.com/elastic/integrations/pull/7042
```
Comment (bhapas) [inline packages/system_audit/changelog.yml:6]:
```suggestion
      link: https://github.com/elastic/integrations/pull/7104
```

**Review by bhapas — COMMENTED** [review]
LGTM. Can we have some system test added / updated to catch this from future updates?

**Comment by efd6** [comment]
Testing this would require adding a filebeat processor to the system config. I've tried that, and the config does not come through correctly (it appears in the rendered config as `- object Object`). I can find no instance in any package of the `processors` var being exercised in tests.

**Comment by andrewkroh** [comment]
In the system test config did you pass the YAML through as a string? Fleet expects these variables to be strings containing YAML (as I have observed through inspecting requests from the browser).

```yaml
data_stream:
  vars:
    processors: |-
      - add_fields:
          fields: ...
```

**Comment by efd6** [comment]
No, I did what seemed like the obviously correct thing to do. I will try that.

**Comment by efd6** [comment]
Thanks, that works. Test now fails without the fix, passes with the fix.

**Review by bhapas — APPROVED** [review]
LGTM

**Pull request merged**